daemon/images/image_import.go

```diff
@@ -57,7 +54,17 @@ func (i *ImageService) ImportImage(src string, repository, os string, tag string
 		}
 	}
 
-	config, err := dockerfile.BuildFromConfig(&container.Config{}, changes, os)
+	// Normalize platform - default to the operating system and architecture if not supplied.
+	if platform == nil {
+		platform = &specs.Platform{}
+	}
+	if platform.Architecture == "" {
+		// TODO platforms.Normalize() is inconsistent: it sets default OS, but does not set default Architecture.
```

Comment:
what case is this?

Reply:
`Parse()` adds the same `runtime.GOARCH` if one is not set in the string so this should not be needed.

Reply:
Right, but that's assuming it always originates from a string that's parsed using that function 🤔, so it's either string -> parse (which also normalises), or `Normalize()` which normalises, but slightly different.